taichi/backends/device.h

```diff
@@ -31,6 +31,103 @@ enum class DeviceCapability : uint32_t {
   vk_has_spv_variable_ptr,
 };
 
+class Device;
+struct DeviceAllocation;
+struct DevicePtr;
+
+// TODO: Figure out how to support images. Temporary solutions is to have all
+// opque types such as images work as an allocation
+struct DeviceAllocation {
```

Comment:
I'm a bit worried about this design: In AOT mode, users likely have a set of buffers. This would prevent them from using their own.

Reply:
We can have the device API support importing external buffers. Assuming in AOT mode the Vulkan runtime is still there

Reply:
Sounds good thx!